### PR TITLE
TDL-19414: Update datatype for some fields

### PR DESCRIPTION
**Pull request by hpatel41**
# Description of change
TDL-19414: Update datatype for some fields
- Updated datatype for fields from the [API Document](https://developer.rechargepayments.com/2021-11)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_recharge/client.py [hunk 1]
```diff
@@ -184,11 +184,10 @@ def request(self, method, path=None, url=None, **kwargs): # pylint: disable=too-
...
         kwargs['headers']['X-Recharge-Access-Token'] = self.__access_token
         kwargs['headers']['Accept'] = 'application/json'
-        # If we did not specify any API Version during API Call, the Recharge will use the default API Version of our store
-        # the 'collections' was added as part of API Version: '2021-11', for older API Version,
-        # we will get empty records so adding 'X-Recharge-Version' for 'collections' API call
-        if path == 'collections':
-            kwargs['headers']['X-Recharge-Version'] = '2021-11'
+        kwargs['headers']['X-Recharge-Version'] = '2021-11'
+        # Products and Shop are not supported in new version: 2021-11
+        if path in ['products', 'shop']:
```
Comment (savan-chovatiya) [inline tap_recharge/client.py:189]: We can remove the `shop` from here as it is replaced with a `store` in another PR.

Comment (harshpatel4crest) [reply tap_recharge/client.py:189]: When we will remove **shop** from here, the Tap will use 2021-11 for getting shop records. We will get 404 Not Found error:
> {
>     "error": "Not Found"
> }

**Review by dbshah1212 — APPROVED** [review]

**Review by savan-chovatiya — APPROVED** [review]

File: tests/unittests/test_request_timeout.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def test_config_provided_request_timeout(self, mock_get, mock_request):
         client = RechargeClient(**config)
         client.request("GET", "dummy_path")
         
-        mock_request.assert_called_with('GET', 'https://api.rechargeapps.com/dummy_path', stream=True, timeout=100.0, headers={'X-Recharge-Access-Token': 'dummy_at', 'Accept': 'application/json', 'User-Agent': 'dummy_ua'})
+        mock_request.assert_called_with('GET', 'https://api.rechargeapps.com/dummy_path', stream=True, timeout=100.0, headers={'X-Recharge-Access-Token': 'dummy_at', 'Accept': 'application/json', 'X-Recharge-Version': '2021-11', 'User-Agent': 'dummy_ua'})
```
Comment (RushiT0122) [inline tests/unittests/test_request_timeout.py:102]: Generalise the headrs in the `mock_request.assert_called_with()` to reduce the line length and improve readability.

**Review by RushiT0122 — CHANGES_REQUESTED** [review]
Requested changes in the `test_request_timeout.py` file.

**Comment by harshpatel4crest** [comment]
> Requested changes in the `test_request_timeout.py` file.

@RushiT0122 That has been covered as part of PR #32

**Review by RushiT0122 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**